tests/SCsub

```diff
@@ -6,9 +6,6 @@ env.tests_sources = []
 
 env_tests = env.Clone()
 
-# Enable test framework and inform it of configuration method.
-env_tests.Append(CPPDEFINES=["DOCTEST_CONFIG_IMPLEMENT"])
-
```

Comment:
I removed this and define `DOCTEST_CONFIG_IMPLEMENT` in `test_macros.cpp` once. Including doctest header twice in source files doesn't protect the implementation to be included once, and leads to linker errors with duplicate symbols.